File: NSpecSpecs/describe_RunningSpecs/describe_example.cs
using System;
using NSpec;
using NUnit.Framework;

namespace NSpecSpecs.WhenRunningSpecs
{
    [TestFixture]
    [Category("RunningSpecs")]
    public class describe_example : when_running_specs
    {
        class SpecClass : nspec
        {
            void it_changes_status_after_run()
            {
            }

            void it_passes()
            {
            }

            void it_fails()
            {
                throw new Exception();
            }
        }

        [Test]
        public void execution_status_changes_after_run()
        {
            Run(typeof(SpecClass));
            var ex = TheExample("it changes status after run");

            //ex.HasRun.should_be_false(); //broken after making init and run happen all at once

            ex.HasRun.should_be_true();
        }

        [Test]
        public void passing_status_is_passed_when_it_succeeds()
        {
            Run(typeof(SpecClass));

            TheExample("it passes").should_have_passed();
        }

        [Test]
        public void passing_status_is_not_passed_when_it_fails()
        {
            Run(typeof(SpecClass));

            TheExample("it fails").should_have_failed();
        }

        {
            void describe_specs_with_anonymous_lambdas()
            {
                context["Some context with anonymous lambdas"] = () =>
                {
                    it["has an anonymous lambda"] = () =>
                    {
                    };
                };
            }
        }

        [Test]
        public void finds_and_runs_three_class_level_examples()
        {
            Run(typeof(SpecClass));

            TheExampleCount().should_be(3);
        }

        [Test]
        public void finds_and_runs_only_one_example_ignoring_anonymous_lambdas()
        {
            Run(typeof(SpecClassWithAnonymouseLambdas));

            TheExampleCount().should_be(1);
        [Test]
        public void finds_and_runs_only_one_example_ignoring_anonymous_lambdas()
        {

            TheExampleCount().should_be(1);
        }
    }
}

File: NSpecSpecsVB/Describe_RunningSpecs/Describe_Example.vb
Imports NSpec
Imports NUnit.Framework
Imports NSpecSpecs
Imports NSpecSpecs.WhenRunningSpecs

Namespace WhenRunningSpecs

    <TestFixture()>
    <Category("RunningSpecs")>
    Public Class DescribeExample
        Inherits when_running_specs

        Public Class SpecClass
            Inherits NSpec.nspec

            Sub It_Changes_Status_After_Run()

            End Sub

            Sub It_Passes()

            End Sub

            Sub It_Fails()
                Throw New Exception()
            End Sub
        End Class

        <Test()>
        Sub Execution_Status_Changes_After_Run()
            Run(GetType(SpecClass))

            Dim ex = TheExample("It Changes Status After Run")

            ex.HasRun.should_be_true()
        End Sub

        <Test()>
        Sub Passing_Status_Is_Passed_When_It_Succeeds()
            Run(GetType(SpecClass))

            TheExample("It Passes").should_have_passed()
        End Sub

        <Test()>
        Sub Passing_Status_Is_Not_Passed_When_It_Fails()
            Run(GetType(SpecClass))

            TheExample("It Fails").should_have_failed()
        End Sub
    End Class

End Namespace

